### PR TITLE
build: fix nit around bazel build at root scope.

### PR DESCRIPTION
**Pull request by htuch**
Previously, if a developer did:

bazel build //...

ci/prebuilt/BUILD would interfere and cause Bazel to be sad. This is fixed by renaming to
ci/prebuilt/BUILD.deps. This reduces developer surprise.

It's still broken in CI to do something like "bazel test //...", since the thirdparty/
and thirdpary_build/ directories may contain BUILD files of their own and are symlinked in.

### CODE REVIEW

**Comment by htuch** [comment]
@PiotrSikora It's fugly but it does fix the point you made about `bazel build //...`. I honestly think it's so fragile we shouldn't commit this PR (the Bazel treatment of external deps is already complex enough), but we can if you and @mattklein123  think it would make things less surprising to developers who haven't been conditions to only expects `//<something>/...` to work.

File: ci/build_setup.sh [hunk 1]
```diff
@@ -101,5 +104,7 @@ function cleanup() {
   rm -f "${ENVOY_CI_DIR}"/bazel-*
   rm -rf "${ENVOY_CI_DIR}"/bazel
   rm -rf "${ENVOY_CI_DIR}"/tools
+  rm -f "${ENVOY_SRCDIR}"/ci/prebuilt/BUILD
+  touch "${ENVOY_SRCDIR}"/ci/prebuilt/BUILD
```
Comment (mattklein123) [inline ci/build_setup.sh]: ? what is this for?

Comment (htuch) [inline ci/build_setup.sh]: Avoid leaving the tree in a dirty state once done, i.e. restore BUILD back to what it was.

**Comment by htuch** [comment]
It won't dirty the repo in https://github.com/lyft/envoy-filter-example/, because it builds using `repositories.bzl`. It will dirty up your repo at Lyft though. I'll add it to the .gitignore and you can decide if it's worth merging or not.

File: ci/build_setup.sh (hunk 1 above)
Comment (mattklein123) [inline ci/build_setup.sh]: Ah OK this confused me, I thought the diff was deleting the file, but it's making it empty. Why do we need the empty BUILD file?

**Comment by mattklein123** [comment]
I don't really have a problem with this if we can add to .gitignore. I'm confused why we need an empty BUILD file, but commented on that above.

**Comment by mattklein123** [comment]
@htuch also, do you think we should be not using the prebuilt approach for our internal build? I thought we had previously agreed that was the best approach? Should we switch to not using the prebuilt stuff and do it like envoy-filter-example?

**Comment by htuch** [comment]
Regarding why we need the empty `BUILD`; `ctxt.symlink(Label("//ci/prebuilt:BUILD.deps"), "BUILD")` can't find `BUILD.deps` unless there is a `BUILD` file in that directory.

**Comment by htuch** [comment]
Feel free to switch your internal build over to non-prebuilt, I think the dev UX is now fairly good, since I added the support for caching in https://github.com/lyft/envoy/pull/780. I think we need to continue to provide support for prebuilts and validate in CI, we've seen demand for this from folks (@mattwoodyard).

**Comment by mattklein123** [comment]
TBH I'm more comfortable using the prebuilts in our internal build. We have everything setup pretty nicely. I'm kind of confused why symlinking a file requires an empty file. Is that a Skylark thing? Is there any way to make this work without dirtying the repo? Otherwise my inclination is to just document somewhere and close?

**Comment by htuch** [comment]
Yeah, since `BUILD.deps` is referenced by the label `//ci/prebuilt:BUILD.deps`, Bazel seems to need a `BUILD` in that directory to know to treat it as a directory it should consider in label evaluation.

@PiotrSikora thoughts?

**Comment by mattklein123** [comment]
@htuch @PiotrSikora what are we thinking on this one? My feeling is it not put in the hack and just maybe document the abnormality somewhere. Thoughts?

**Comment by PiotrSikora** [comment]
I think you should land this.

The whole `//ci/prebuilt` is a hack, IMHO, so adding this to make `//...` work is definitely OK.

**Comment by mattklein123** [comment]
This will not work for our internal build. Is there any solution?

**Comment by htuch** [comment]
Dropping since I don't have a clean way to handle this without dirtying up the tree for others.

**Pull request closed without merging**